src/zenml/config/step_configurations.py

```diff
@@ -26,14 +28,30 @@
 class PartialArtifactConfiguration(StrictBaseModel):
     """Class representing a partial input/output artifact configuration."""
 
-    artifact_source: Optional[str] = None
     materializer_source: Optional[str] = None
 
+    @root_validator(pre=True)
+    def _remove_deprecated_attributes(
+        cls, values: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Removes deprecated attributes from the values dict.
+
+        Args:
+            values: The values dict used to instantiate the model.
+
+        Returns:
+            The values dict without deprecated attributes.
+        """
+        deprecated_attributes = ["artifact_source"]
+        for deprecated_attribute in deprecated_attributes:
+            if deprecated_attribute in values:
+                values.pop(deprecated_attribute)
```

Comment:
I prefer this one-liner here, up to you what you think is clearer

```suggestion
            values.pop(deprecated_attribute, None)
```

Reply:
I'll stay with the explicit version for this one